orthoinference/src/main/java/org/reactome/orthoinference/InferEvents.java

```diff
@@ -0,0 +1,306 @@
+package org.reactome.orthoinference;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Properties;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.GKSchemaAttribute;
+import org.gk.schema.SchemaClass;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.reactome.release.common.database.InstanceEditUtils;
+
+/**
+ *
+ * @author jcook
+ * 
+ * The Java version of infer_events.pl -- The gist of this module is that it looks at all existing Human ReactionlikeEvent (RlE) instances (mostly Reactions and BlackBoxEvents) in the Test_Reactome database,
+ * and attempts to computationally infer them in each of Reactome's model organisms. Each RlE is broken down into its primary components (input, output, catalyst, and regulator), which are themselves broken
+ * into their PhysicalEntity subunits. The homology data used for the inference process currently comes from Ensembl Compara and is generated during the 'Orthopairs' step of the Reactome release process.
+ * After all inference attempts for each RlE has been completed in an organism, the pathways that contain the reactions are filled with these newly inferred ones. 
+ * 
+ *
+ */
+
+// To Reviewers: I recommend following the code sequentially, rather then looking at each class separately.
+public class InferEvents
+{
+	static MySQLAdaptor dbAdaptor = null;
+	private static GKInstance instanceEdit;
+	private static GKInstance speciesInst;
+	private static HashMap<GKInstance,GKInstance> manualEventToNonHumanSource = new HashMap<GKInstance,GKInstance>();
```

Comment:
All Collections (Maps, Lists, Sets, etc.) are better to be declared with the interface.  It might be good to do a search of all files for the declarations and change them to use the interfaces rather than implementations like HashMap and ArrayList.

Reply:
I'm not sure I understand the concept of Interfaces still, to be honest

Reply:
The main idea is that an interface only specifies what methods a class will have, but not how they will do things.  This way any code that implements the interface is agreeing to a contract that it will provide the methods and as a client, you and other developers know that you can call a function and get back a value (or not if the return is void).  This lets you know that you can do things with code the uses the interface but if the algorithm doing them changes under the hood, it won't require the code that uses it to change because the interaction is still the same from outside.  Using an interface as the declaring type also only permits you to use methods specified by the interface whereas the implementation allows you to use things that may be specific only to that class.  If you use those kinds of methods, it ties you to using that specific implementation and doesn't let you switch if you decide something else might be better or more appropriate later.

All that said, it may be no problem to use a specific implementation but if there's not a reason that you need to (like a method specific only to that implementation), sticking with the interface is likely better.

Some good discussion about it here:
https://stackoverflow.com/questions/9852831/polymorphism-why-use-list-list-new-arraylist-instead-of-arraylist-list-n